frontend/src/pages/history.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import Link from 'next/link';
+import RouteGuard from '@/components/RouteGuard';
+import { Role } from '@/utils/constants';
+import { useAuthContext } from '@/contexts/AuthContext';
+
+const submission_service_port = 5433;
+
+export default function SubmissionHistory() {
+  const { user } = useAuthContext();
+
+  const [submissions, setSubmissions] = useState([]);
+
+  useEffect(() => {
+    if (user == null)
+      return;
+    const fetchSubmissionHistory = async () => {
+      try {
+        console.log("Username: " + user.username);
+        const response = await fetch(`http://localhost:${submission_service_port}/user/${user.username}`);
+        if (!response.ok)
+          throw new Error(`HTTP error! status: ${response.status}`);
+        const data = await response.json();
+        setSubmissions(data);
+      } catch (error) {
+        console.error('Error fetching submission history:', error);
+      }
+    };
+    fetchSubmissionHistory();
+  }, [user]);
+
+  return (
+    <RouteGuard allowedRoles={[Role.USER, Role.ADMIN]}>
+      <div>
```

Comment:
I've been using MaterialUI components for the frontend, so I think it would be good to keep it consistent (though I would be happy to modify it)

Reply:
I'll leave that part to you